From 2eb78e46430c4e98d9e581d663a28c358c481209 Mon Sep 17 00:00:00 2001
From: Hannes Frederic Sowa <hsowa@redhat.com>
Date: Tue, 3 Jun 2014 13:23:47 -0400
Subject: [lib] random32: use msecs_to_jiffies for reseed timer

Message-id: <1401801829-9802-9-git-send-email-hsowa@redhat.com>
Patchwork-id: 82391
O-Subject: [RHEL7.1 PATCH net 08/10] random32: use msecs_to_jiffies for reseed timer
Bugzilla: 1100299
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Florian Westphal <fwestpha@redhat.com>
RH-Acked-by: Daniel Borkmann <dborkman@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1100299
Upstream Status: 0125737accc5aac532719aecd80615364caa9e0f
Brew: http://brewweb.devel.redhat.com/brew/taskinfo?taskID=7530240
Tested: verified with prandom kernel test suite

Upstream Commit:
commit 0125737accc5aac532719aecd80615364caa9e0f
Author: Daniel Borkmann <dborkman@redhat.com>
Date:   Tue Nov 12 23:45:42 2013 +0100

    random32: use msecs_to_jiffies for reseed timer

    Use msecs_to_jiffies, for these calculations as different HZ
    considerations are taken into account for conversion of the timer
    shot, and also it makes the code more readable.

    Signed-off-by: Daniel Borkmann <dborkman@redhat.com>
    Signed-off-by: Hannes Frederic Sowa <hannes@stressinduktion.org>
    Signed-off-by: David S. Miller <davem@davemloft.net>

Signed-off-by: Hannes Frederic Sowa <hsowa@redhat.com>
Signed-off-by: Jiri Benc <jbenc@redhat.com>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/lib/random32.c b/lib/random32.c
index 4f9d5df..1e5b2df 100644
--- a/lib/random32.c
+++ b/lib/random32.c
@@ -214,18 +214,22 @@ static DEFINE_TIMER(seed_timer, __prandom_timer, 0, 0);
 static void __prandom_timer(unsigned long dontcare)
 {
  u32 entropy;
+ unsigned long expires;
 
  get_random_bytes(&entropy, sizeof(entropy));
  prandom_seed(entropy);
+
  /* reseed every ~60 seconds, in [40 .. 80) interval with slack */
- seed_timer.expires = jiffies + (40 * HZ + (prandom_u32() % (40 * HZ)));
+ expires = 40 + (prandom_u32() % 40);
+ seed_timer.expires = jiffies + msecs_to_jiffies(expires * MSEC_PER_SEC);
+
  add_timer(&seed_timer);
 }
 
 static void __init __prandom_start_seed_timer(void)
 {
  set_timer_slack(&seed_timer, HZ);
- seed_timer.expires = jiffies + 40 * HZ;
+ seed_timer.expires = jiffies + msecs_to_jiffies(40 * MSEC_PER_SEC);
  add_timer(&seed_timer);
 }
 
-- 
1.7.1